packages/@n8n/nodes-langchain/nodes/chains/InformationExtractor/InformationExtractor.node.ts

```diff
@@ -0,0 +1,307 @@
+import { jsonParse, NodeConnectionType, NodeOperationError } from 'n8n-workflow';
+import type {
+	INodeType,
+	INodeTypeDescription,
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodePropertyOptions,
+} from 'n8n-workflow';
+import type { JSONSchema7 } from 'json-schema';
+import type { BaseLanguageModel } from '@langchain/core/language_models/base';
+import { ChatPromptTemplate, SystemMessagePromptTemplate } from '@langchain/core/prompts';
+import type { z } from 'zod';
+import { OutputFixingParser, StructuredOutputParser } from 'langchain/output_parsers';
+import { HumanMessage } from '@langchain/core/messages';
+import { generateSchema, getSandboxWithZod } from '../../../utils/schemaParsing';
+import {
+	inputSchemaField,
+	jsonSchemaExampleField,
+	schemaTypeField,
+} from '../../../utils/descriptions';
+import { getTracingConfig } from '../../../utils/tracing';
+import type { AttributeDefinition } from './types';
+import { makeZodSchemaFromAttributes } from './helpers';
+
+const SYSTEM_PROMPT_TEMPLATE = `You are an expert extraction algorithm.
+Only extract relevant information from the text.
+If you do not know the value of an attribute asked to extract, you may omit the attribute's value.`;
```

Comment:
Does this work with things that are marked as required in the attribute field?

Reply:
It works. I guess this phrase is there to reduce hallucinations. This is the prompt from LangChain's information extraction example. I haven't done much prompt engineering, but it seems the feature extraction schema part has much more weight in the prompt.